examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -23,412 +25,306 @@ import TerminalLoggers
-        q3 = 0.1
-    end
-    q4 = 1 - 3 / 10 * (q1 + q2 + q3)
+    k = 10 * R / τ
+    λp = λ - FT(360) * t / τ
 
-    ρq1 = ρ_ref(z) * q1
-    ρq2 = ρ_ref(z) * q2
-    ρq3 = ρ_ref(z) * q3
-    ρq4 = ρ_ref(z) * q4
+    ua =
+        k * sind(λp)^2 * sind(2 * ϕ) * cos(FT(π) * t / τ) +
+        2 * FT(π) * R / τ * cosd(ϕ)
+    ud =
+        ω_0 * R / b / p_top *
+        cosd(λp) *
+        cosd(ϕ)^2 *
+        cos(2 * FT(π) * t / τ) *
+        (-exp((p(z) - p_0) / b / p_top) + exp((p_top - p(z)) / b / p_top))
+    uu = ua + ud
 
-    return (
-        ρ = ρ_ref(z),
-        ρq1 = ρq1,
-        ρq2 = ρq2,
-        ρq3 = ρq3,
-        ρq4 = ρq4,
-        ρq1_td = ρq1,
-        ρq2_td = ρq2,
-        ρq3_td = ρq3,
-        ρq4_td = ρq4,
-    )
+    uv = k * sind(2 * λp) * cosd(ϕ) * cos(FT(π) * t / τ)
+
+    sp =
+        1 + exp((p_top - p_0) / b / p_top) - exp((p(z) - p_0) / b / p_top) -
+        exp((p_top - p(z)) / b / p_top)
+    ω = ω_0 * sind(λp) * cosd(ϕ) * cos(2 * FT(π) * t / τ) * sp
+    uw = -ω / ρ_ref(z) / grav
+
+    Geometry.Covariant123Vector(Geometry.UVWVector(uu, uv, uw), local_geometry)
 end
 
-y0 = Fields.FieldVector(
-    ρ = y0.ρ,
-    ρq1 = y0.ρq1,
-    ρq2 = y0.ρq2,
-    ρq3 = y0.ρq3,
-    ρq4 = y0.ρq4,
-    ρq1_td = y0.ρq1,
-    ρq2_td = y0.ρq2,
-    ρq3_td = y0.ρq3,
-    ρq4_td = y0.ρq4,
-)
+# "Tendency increment", which computes Y2 .+= dt * ∂Y/∂t(Y1, t)
+function increment!(Y2, Y1, cache, t, dt, beta)
+    (; cent_uvw, face_uvw, face_uv, face_w, limiter, Δₕq, fct_op, ρq_td) = cache
+    ρ1 = Y1.c.ρ
+    ρq1 = Y1.c.ρq
+    ρ2 = Y2.c.ρ
+    ρq2 = Y2.c.ρq
+
+    cent_local_geometry = Fields.local_geometry_field(cent_uvw)
+    face_local_geometry = Fields.local_geometry_field(face_uvw)
+    @. cent_uvw = local_flow(cent_local_geometry, t)
+    @. face_uvw = local_flow(face_local_geometry, t)
+    horz_axis = Ref(Geometry.Covariant12Axis())
+    vert_axis = Ref(Geometry.Covariant3Axis())
+    @. face_uv = Geometry.project(horz_axis, face_uvw)
+    @. face_w = Geometry.project(vert_axis, face_uvw)
+
+    # NOTE: With a limiter, the order of operations 1-5 below matters!
+
+    # 1) Horizontal transport:
+    @. ρ2 -= dt * hdiv(ρ1 * cent_uvw)
+    for n in 1:5 # TODO: update RecursiveApply/Operators to eliminate this loop
+        ρq1_n = ρq1.:($n)
+        ρq2_n = ρq2.:($n)
+        @. ρq2_n -= dt * hdiv(ρq1_n * cent_uvw)
+    end
 
-function rhs!(dydt, y, parameters, t, alpha, beta)
+    # 2) Limiter application:
+    if !isnothing(limiter)
+        Limiters.compute_bounds!(limiter, ρq1, ρ1)
+        Limiters.apply_limiter!(ρq2, ρ2, limiter)
+    end
 
-    (; τ, coords, face_coords, ystar, y_td) = parameters
+    # 3) Hyperdiffusion of tracers:
+    @. Δₕq = hwdiv(hgrad(ρq1 / ρ1))
+    Spaces.weighted_dss!(Δₕq)
+    @. ρq2 -= dt * D₄ * hwdiv(ρ1 * hgrad(Δₕq))
+
+    # 4) Vertical transport:
+    @. ρ2 -= dt * vdivf2c(Ic2f(ρ1) * face_uvw)
+    for n in 1:5 # TODO: update RecursiveApply/Operators to eliminate this loop
```

Comment:
Is this the same issue as #724 ?

Reply:
Not quite; there's a lot of additional things that need to be patched. I managed to eliminate the first loop by extending `rmul` to work with `AxisTensor`s, but then the second loop required way more modifications, including to `divergence_result_type`, `Jcontravariant3`, and probably a bunch of other stuff I would have found if I'd kept trying. For now, I'll just leave these as loops.